src/io/benchmark.zig

```diff
@@ -2,237 +2,237 @@ const std = @import("std");
-                &std.mem.toBytes(@as(c_int, 1)),
-            );
-            try os.bind(self.server.fd, &address.any, address.getOsSockLen());
-            try os.listen(self.server.fd, 1);
-
-            // Start accepting the client
-            self.io.accept(
-                *Context,
-                &self,
-                on_accept,
-                &self.server.completion,
-                self.server.fd,
-            );
-
-            // Setup the client connection
-            self.tx.socket.fd = try IO.openSocket(os.AF_INET, os.SOCK_STREAM, os.IPPROTO_TCP);
-            defer os.closeSocket(self.tx.socket.fd);
-
-            self.io.connect(
-                *Context,
-                &self,
-                on_connect,
-                &self.tx.socket.completion,
-                self.tx.socket.fd,
-                address,
-            );
-
-            // Run the IO loop, calling either tick() or run_for_ns() at "pseudo-random"
-            // to benchmark each io-driving execution path
-            var tick: usize = 0xdeadbeef;
-            while (self.isRunning()) : (tick +%= 1) {
-                if (tick % 61 == 0) {
-                    const timeout_ns = tick % (10 * std.time.ns_per_ms);
-                    try self.io.run_for_ns(@intCast(u63, timeout_ns));
-                } else {
-                    try self.io.tick();
-                }
-            }
+// 1 MB: larger than socket buffer so forces io_pending on darwin
+// Configure this value to smaller amounts to test IO scheduling overhead
+const buffer_size = 1 * 1024 * 1024;
 
-            // Assert that everything is connected
-            try assert(self.server.fd != -1);
-            try assert(self.tx.socket.fd != -1);
-            try assert(self.rx.socket.fd != -1);
+// max time for the benchmark to run
+const run_duration = 1 * std.time.ns_per_s;
 
-            // Close the accepted client socket.
-            // The actual client socket + server socket are closed by defer
-            os.closeSocket(self.rx.socket.fd);
+pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    const allocator = &gpa.allocator;
+    defer {
+        const leaks = gpa.deinit();
+        assert(!leaks);
+    }
+
+    const buffer = try allocator.alloc(u8, buffer_size * 2);
+    defer allocator.free(buffer);
+    std.mem.set(u8, buffer, 0);
+
+    var timer = Time{};
+    const started = timer.monotonic();
+    var self = Context{
+        .io = try IO.init(32, 0),
+        .timer = &timer,
+        .started = started,
+        .current = started,
+        .tx = .{ .buffer = buffer[0 * buffer_size ..][0..buffer_size] },
+        .rx = .{ .buffer = buffer[1 * buffer_size ..][0..buffer_size] },
+    };
+
+    defer {
+        self.io.deinit();
+        const elapsed_ns = self.current - started;
+        const transferred_mb = @intToFloat(f64, self.transferred) / 1024 / 1024;
+
+        std.debug.warn("IO throughput test: took {}ms @ {d:.2} MB/s\n", .{
```

Comment:
We should use `std.debug.print` here (or a stdout writer) instead since `warn` is deprecated?